packages/edit-widgets/src/blocks/legacy-widget/index.js

```diff
@@ -34,49 +40,74 @@ export const settings = {
  * @return {Object} Block object.
  */
 export const create = ( editorSettings ) => {
-	const legacyWidgets = editorSettings?.availableLegacyWidgets ?? {};
+	const legacyWidgets = Object.entries(
+		editorSettings.availableLegacyWidgets
+	).filter( ( [ , widget ] ) => ! widget.isHidden );
+
+	return [
+		{
+			...blockBasis,
+			settings: {
+				...blockBasis.settings,
+				variations: legacyWidgets
```

Comment:
Should we go all in with the _block per widget_ approach so that there's only one type of block to test and debug?